### PR TITLE
Add ability to remove mounts and maps.

### LINKED ISSUE #91: No option to remove an already defined autofs::mount
Dear developers, 

today I just wanted to remove  an existing autofs::mount, but, there is no _ensure => absent_ option available. The only workaround I found was to set mapcontents => [], but this doesn't remove the entry in auto.master and also the mapfile is not removed. 

Would be possible to extend the module having in mind also the removal for already defined autofs::mount entries  ?

Thank you
Dan

### PR DESCRIPTION
**Pull request by dhollinger**
closes #91 

Important note: This will remove entire `autofs::map` mapfiles. 

If you want to remove just the content of a `mapfile` either through an `autofs::mount` or `autofs::map`, simply remove the content from the `mapcontents` parameter in either defined type.

### CODE REVIEW

File: manifests/map.pp [hunk 1]
```diff
@@ -27,12 +28,13 @@
...
 define autofs::map (
-  Array $mapcontents,
-  Stdlib::Absolutepath $mapfile,
+  Enum['present', 'absent'] $ensure                                 = 'present',
+  Stdlib::Absolutepath $mapfile                                     = $title,
   Enum['autofs/auto.map.erb', 'autofs/auto.map.exec.erb'] $template = 'autofs/auto.map.erb',
   String $mapmode                                                   = '0644',
   Boolean $replace                                                  = true,
   Integer $order                                                    = 1,
+  Optional[Array] $mapcontents                                      = undef,
```
Comment (bastelfreak) [inline manifests/map.pp]: as discussed on IRC, this should probably be Variant[String,Array]

Comment (dhollinger) [reply manifests/map.pp]: Probably should be `Optional[Variant[String, Array]]` as it is an optional value

Comment (dhollinger) [reply manifests/map.pp]: I couldn't find the reason why it was optional. Going to do a quick over on the code that generates and removes mounts and maps to make sure we don't make it overly difficult to remove mounts/maps with an `ensure => absent`

File: manifests/map.pp [hunk 2]
```diff
@@ -27,12 +28,13 @@
...
 define autofs::map (
-  Array $mapcontents,
-  Stdlib::Absolutepath $mapfile,
+  Enum['present', 'absent'] $ensure                                 = 'present',
+  Stdlib::Absolutepath $mapfile                                     = $title,
   Enum['autofs/auto.map.erb', 'autofs/auto.map.exec.erb'] $template = 'autofs/auto.map.erb',
   String $mapmode                                                   = '0644',
   Boolean $replace                                                  = true,
   Integer $order                                                    = 1,
+  Variant[Array, String] $mapcontents                               = [],
```
Comment (bastelfreak) [inline manifests/map.pp]: Datatypes \o/

**Pull request merged**